x-pack/legacy/plugins/epm/public/screens/add_data_source/index.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import { ICON_TYPES, EuiPageHeader, EuiTitle, EuiFlexGroup, EuiIcon, EuiPanel } from '@elastic/eui';
+import styled from 'styled-components';
+import { NavButtonBack } from '../../components/nav_button_back';
+import { useLinks } from '../../hooks';
+import { PackageInfo } from '../../../common/types';
+import { getPackageInfoByKey } from '../../data';
+import { AddDataSourceSteps } from './add_data_source_steps';
+
+export interface AddDataSourceProps {
+  pkgkey: string;
+}
+
+// TODO: change to percentages?
+const SIDEBAR_WIDTH = '168px';
+
+const PageContainer = styled.div`
+  margin: 0 auto;
+  width: 1200px;
+`;
+const PageBody = styled.div`
+  min-width: 772px;
+`;
+const NavButtonBackWrapper = styled.div`
+  padding: ${props => props.theme.eui.paddingSizes.xl} 0 ${props => props.theme.eui.paddingSizes.m}
+    0;
+`;
+const SideBar = styled.div`
+  margin-right: ${props => props.theme.eui.paddingSizes.xl};
+  max-width: ${SIDEBAR_WIDTH};
+`;
+const IconPanel = styled(EuiPanel)`
+  padding: ${props => props.theme.eui.paddingSizes.xl} !important;
+  text-align: center;
+  width: ${SIDEBAR_WIDTH};
+  height: ${SIDEBAR_WIDTH};
+`;
+export function AddDataSource({ pkgkey }: AddDataSourceProps) {
+  const [info, setInfo] = useState<PackageInfo | null>(null);
+  const { toDetailView } = useLinks();
+
+  useEffect(() => {
+    getPackageInfoByKey(pkgkey).then(response => {
+      setInfo({ ...response });
+    });
+  }, [pkgkey]);
+
+  // don't have designs for loading/empty states
+  if (!info) return null;
+
+  const { version, name, title } = info;
+  const iconType = ICON_TYPES.find(key => key.toLowerCase() === `logo${name}`);
+
+  return (
+    <PageContainer>
+      <NavButtonBackWrapper>
+        <NavButtonBack
+          href={toDetailView({ name, panel: 'data-sources', version })}
+          text="Cancel"
+        />
+      </NavButtonBackWrapper>
+      <EuiFlexGroup gutterSize="none">
+        <SideBar>
+          <IconPanel>
```

Comment:
I believe this will render and empty `IconPanel` if there's no `iconType`

Reply:
yes, I was thinking we'd have some kind of default image instead of an empty space at some point.

Reply:
That's inconsistent with what we do on the details page. We can ship but I'd like @hbharding to say what the desired behavior is and come back and correct it off need be.

Reply:
Initially the API served a default icon but it felt odd that the API component makes the decision which is the default icon. I would like to see a default icon but be part of the UI somewhere.

Reply:
Henry said he has a design with a default icon so we can add that in the other issue as well